apps/test/unit/storage/FirebaseStorageTest.js

```diff
@@ -160,6 +160,27 @@ describe('FirebaseStorage', () => {
           });
       }
     });
+    it('escapes periods in the key', done => {
+      let didWarn = false;
+      FirebaseStorage.setKeyValue(
+        'foo.bar',
+        'baz',
+        () => verifyValue(),
+        err => {
+          didWarn = true;
```

Comment:
are you able to confirm that this test will fail if this error handler is called? I see we are setting didWarn, but the only place we check its value appears to be in the success handler, which I wouldn't expect to be called if the error handler is called.

Reply:
thanks for catching this, I updated the test and confirmed it passes when we escape the periods, and fails otherwise. `MockFirebase` seems to actually allow periods in the key name, so the test only fails because `expect(snapshot.val()).to.deep.equal({'foo%2Ebar': '"baz"'});` isn't true, but I think that's okay.

![image](https://user-images.githubusercontent.com/8787187/85622772-31427580-b61c-11ea-9d9d-071224594602.png)

![image](https://user-images.githubusercontent.com/8787187/85622747-28ea3a80-b61c-11ea-8ea1-00ff90fcb6b4.png)